ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_changing_kind_of_array.py

```diff
@@ -0,0 +1,42 @@
+# local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+from hypothesis import strategies as st
+
+
+# asanyarray
+@handle_frontend_test(
+    fn_tree="numpy.asanyarray",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("valid")),
+    dtype=helpers.get_dtypes("valid", full=False),
+    order=st.sampled_from(["C", "F", "A", "K"]),
+    like=helpers.get_dtypes("valid"),
```

Comment:
`like` is generating dtypes, not arrays (you are using the same helper strategy as for `dtype`. See [this list of strategies](https://lets-unify.ai/ivy/deep_dive/ivy_tests.html#why-do-we-need-helper-functions) and try out ones that generate arrays.